### ISSUE TITLE
Updating to v3.3.10 causes infinite loop for lazy query

### ISSUE THREAD

**Issue by emmenko** [apollographql/apollo-client#7713]
**Intended outcome:**

As per title, updating to the latest version (`3.3.10`) should not cause any regression.

**Actual outcome:**

From what I was able to debug, when using `useLazyQuery` it seems that `called` is always `false`, causing the function to trigger the query to be called again and again.

Eventually we get [this error](https://reactjs.org/docs/error-decoder.html/?invariant=301) from React

```
Too many re-renders. React limits the number of renders to prevent an infinite loop.
```

**How to reproduce the issue:**

If necessary, I'll try to create a reproduction. I wanted to report the issue first in case it's some known problem.

**Comment by emmenko** [comment]
Here's the reproduction case: https://codesandbox.io/s/repro-apollo-issue-7713-jsn9u

**Comment by benjamn** [comment]
@emmenko This should be fixed in `@apollo/client@3.3.11`, at least in the sense that it restores the v3.3.9 behavior (no infinite loop in your reproduction). As you probably are aware, your real code probably needs some additional logic to handle `loading: true` before using `data`, and React's `<StrictMode/>` renders components twice, which may be surprising. Let us know if you notice any other deviations from previous behavior, or any other surprising aspects of that behavior. Thanks!
